### DISCUSSION TITLE
[A-L] (14 marzo 2024) Esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#18] [Esercitazioni A-L]
In un grafo **non diretto** $G$ un **cammino Hamiltoniano** è un cammino $P$ di $G$ che passa per ogni vertice di $G$ esattamente una volta. Provare o confutare la seguente affermazione: $G$ contiene un cammino Hamiltoniano $\iff$ può essere prodotto un albero di ricerca di $G$ che è un cammino.

Domanda bonus: che succede se invece del cammino Hamiltoniano, si ha un **ciclo Hamiltoniano** in $G$?

**Comment by Frisbon** [comment]
se qualcuno ha soluzione, ping me pls
